common/src/buffer.c

```diff
@@ -0,0 +1,667 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
I don't think that we actually use BUFFER anywhere, we just use CONSTBUFFER/CONSTBUFFER_ARRAY. @dcristo, thoughts on whether we should have just constbuffer here? Or also add buffer? What about constbuffer array? Only CONSTBUFFER is used right now in the file module.

Reply:
I was thinking about this too, I think we should remove buffer.c and CONSTBUFFER_CreateFromBuffer since we do not use it.
We discussed today of adding a BUFFER and CONSTBUFFER_CreateFromBuffer that would simply pick it up without any memory allocation, but I believe that requires reworking the BUFFER anyway.
So for now rather than dragging it all over we should probably just remove it.

---
In reply to: [448497048](https://github.com/Azure/azure-c-pal/pull/7#discussion_r448497048) [](ancestors = 448497048)

Reply:
Btw, I am not very very happy about having constbuffer in the PAL. That is because I do not think it really belongs in here. Maybe we should think a bit about how we can make that happen, but I guess it is for another rainy day.

---
In reply to: [451980764](https://github.com/Azure/azure-c-pal/pull/7#discussion_r451980764) [](ancestors = 451980764,448497048)